src/twisted/web/_flatten.py

```diff
@@ -157,6 +157,25 @@ def _getSlotValue(name, slotData, default=None):
         raise UnfilledSlot(name)
 
 
+def _fork(d):
+    """
+    Create a new L{Deferred} based on C{d} that will fire and fail with C{d}'s
+    result or error, but will not modify C{d}'s callback type.
+    """
+    d2 = Deferred()
+
+    def callback(result):
+        d2.callback(result)
+        return result
+
+    def errback(failure):
+        d2.errback(failure)
+        return failure
```

Comment:
Thanks codecov. This uncovered line actually illustrates a really significant coverage gap in the previous behavior. I might have to do a test-refactoring PR first to fix this.